libraries/testing/tester.cpp

```diff
@@ -561,8 +561,12 @@ namespace eosio::testing {
          // otherwise the testing_allow_voting would be set to true when `i >= 4` even though the user of
          // `produce_blocks` wants it to be true.
          // This is 4 instead of 3 because the extra block has to be produced to log_irreversible
-         if (allow_voting_originally && n > 4)
-            control->testing_allow_voting(i >= n - 4);
+         if (allow_voting_originally && n > 4) {
```

Comment:
Since we vote on the first two and the last four.
```suggestion
         if (allow_voting_originally && n > 6) {
```